apps/api-extractor/src/collector/Collector.ts

```diff
@@ -147,6 +158,57 @@ export class Collector {
   }
 
   /**
+   * Searches the provided package.json for dependencies that match the provided package names and/or glob patterns
+   * in `bundledPackages`.
+   * @param bundledPackages - The list of package names and/or glob patterns to search for in the package.json.
+   * @param packageJson - The package.json of the package being processed.
+   * @returns The set of matching package names.
+   */
+  private static _resolveBundledPackagePatterns(
+    bundledPackages: string[],
+    packageJson: INodePackageJson | undefined
```

Comment:
The theory behind this design is that the glob wildcard is matched against names that appear in **package.json** dependencies.  So what happens if an indirect dependency has a matching name?  Does the implementation distinguish those two cases?


- `my-lib/package.json` depends on `dep-a` and `dep-b`
- `my-lib/node_modules/dep-a/dist/index.d.ts`  imports `dep-b`
- `my-lib/node_modules/dep-b/dist/index.d.ts` imports `dep-c`
- `my-lib/node_modules/dep-c/dist/index.d.ts` imports `dep-b` and but gets a different version
- `my-lib/node_modules/dep-c/node_modules/dep-b/dist/index.d.ts` (probably a preexisting issue?)

Then suppose we have `bundledPackages: [ "dep-*" ]`.

Reply:
Sounds like:
1. The intended behavior would be to match `dep-a` and top-level `dep-b` only
2. Current implementation probably double-bundles `dep-b` but that is a preexisting issue

Reply:
In the old design:  The `bundledPackages` was an exhaustive list of exact strings, and then modules would get rolled up only if they were encountered during `import` module resolution AND matched one of those strings.  This included duplicates like `dep-b` above.

In the new design: We could say `bundledPackages` is a list that only matches dependencies that can be directly resolved from `my-lib`, and if there is a glob `*` it is applied against names that appear in **package.json** and no other names. In other words the glob is a shorthand, and you can always explicitly specify all names.

A different interpretation:  We could alternatively design it such that IF a name is specified in `bundledPackages` AND it does not match any name in **package.json** then that name would be ignored.

Reply:
Based on our discussion, will update to preserve existing, package.json-agnostic behavior for exact package names and continue to resolve wildcards against package.json contents. Will also update docs to call out the distinction.